src/Workspaces/MSBuild/Core/Rpc/RemoteBuildHost.cs

```diff
@@ -33,6 +33,13 @@ public async Task<RemoteProjectFile> LoadProjectFileAsync(string projectFilePath
         return new RemoteProjectFile(_client, remoteProjectFileTargetObject);
     }
 
+    public async Task<RemoteProjectFile> LoadProjectAsync(string projectFilePath, string projectContent, string languageName, CancellationToken cancellationToken)
```

Comment:
Consider adding a comment that the projectFilePath is only a conceptual path here and doesn't need to exist on disk.